common/src/main/java/com/wynntils/mc/mixin/ScreenMixin.java

```diff
@@ -20,8 +22,30 @@
 
 @Mixin(Screen.class)
 public abstract class ScreenMixin {
+    @Final
     @Shadow
-    protected abstract <T extends GuiEventListener & Widget & NarratableEntry> T addRenderableWidget(T widget);
+    private List<GuiEventListener> children;
+
+    @Final
+    @Shadow
+    private List<NarratableEntry> narratables;
+
+    @Final
+    @Shadow
+    private List<Widget> renderables;
+
+    // Making this public is required for the mixin, use this with caution anywhere else
```

Comment:
Is that really so? Sponge normally just uses the mixin classes as a holder for writing Java code, and pays no respect to access modifyers. That's why classes, methods and fields are typically declared abstract or private.

(But even if this can't be made private for some reason, the class can't be instantiated anyway since it's abstract, so I doubt this comment is really needed.)

Reply:
This happens only on fabric iirc, I tried making them protected, it does not work, but when public it works. Not sure if the comment is really needed though.

Reply:
```
[13:29:20] [Client Shutdown Thread/ERROR]: Mixin apply for mod wynntils failed wynntils.mixins.json:ScreenMixin -> net.minecraft.client.gui.screens.Screen: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException PRIVATE overwrite method addRenderableWidget in wynntils.mixins.json:ScreenMixin cannot reduce visibiliy of PUBLIC target method
org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: PRIVATE overwrite method addRenderableWidget in wynntils.mixins.json:ScreenMixin cannot reduce visibiliy of PUBLIC target method
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.conformVisibility(MixinPreProcessorStandard.java:479) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachMethod(MixinPreProcessorStandard.java:590) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachMethods(MixinPreProcessorStandard.java:356) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attach(MixinPreProcessorStandard.java:299) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.createContextFor(MixinPreProcessorStandard.java:277) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinInfo.createContextFor(MixinInfo.java:1289) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:292) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) [sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) [sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) [sponge-mixin-0.11.2+mixin.0.8.5.jar:0.11.2+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:247) [fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:150) [fabric-loader-0.13.3.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:155) [fabric-loader-0.13.3.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) [?:?]
	at net.minecraft.client.main.Main$2.run(Main.java:176) [minecraft-merged-named.jar:?]

```

Reply:
Right, this is not a mixin that is injected into an existing method. 

What are we even trying to do here? Override an existing method to replace it with our own? Create a support method for our needs that happens to collide with an existing method?

Reply:
This is needed so we can add the Wynncraft button to the title screen

Reply:
Ok, fine, it's probably harmless. I'm getting to tired to fully understand what this does. I'll look into it another time.